clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/AbstractVeniceMapper.java

```diff
@@ -138,33 +139,33 @@ protected boolean process(
     if (compressionMetricCollectionEnabled) {
       byte[] compressedRecordValue;
       for (CompressionStrategy compressionStrategy: CompressionStrategy.values()) {
-        if (compressor[compressionStrategy.getValue()] != null) {
-          switch (compressionStrategy) {
-            case NO_OP:
-              // incrTotalUncompressedValueSize() already collected this data
-              break;
-
-            case GZIP:
-            case ZSTD_WITH_DICT:
-              try {
-                if (compressionStrategy == this.compressionStrategy) {
-                  // Extra check to not redo compression
-                  compressedRecordValue = finalRecordValue;
-                } else {
-                  compressedRecordValue = compressor[compressionStrategy.getValue()].compress(recordValue);
-                }
-                if (compressionStrategy == GZIP)
-                  MRJobCounterHelper.incrTotalGzipCompressedValueSize(reporter, compressedRecordValue.length);
-                else
-                  MRJobCounterHelper.incrTotalZstdCompressedValueSize(reporter, compressedRecordValue.length);
-              } catch (IOException e) {
-                // TODO: ignore for now?
-              }
-              break;
-
-            default: // defensive check
-              throw new VeniceException(
-                  "Support for compression Strategy: " + compressionStrategy.name() + " needs to be added");
+        // incrTotalUncompressedValueSize() already collected data for NO_OP
+        if (compressionStrategy != NO_OP && compressor[compressionStrategy.getValue()] != null) {
+          try {
+            if (compressionStrategy == this.compressionStrategy) {
+              // Extra check to not redo compression
+              compressedRecordValue = finalRecordValue;
+            } else {
+              compressedRecordValue = compressor[compressionStrategy.getValue()].compress(recordValue);
+            }
+            switch (compressionStrategy) {
+              case GZIP:
+                MRJobCounterHelper.incrTotalGzipCompressedValueSize(reporter, compressedRecordValue.length);
+                break;
+
+              case ZSTD_WITH_DICT:
+                MRJobCounterHelper.incrTotalZstdCompressedValueSize(reporter, compressedRecordValue.length);
+                break;
+
+              default: // defensive check
```

Comment:
Will `ZSTD` fall under this and trigger an exception and eventual `WARN` log?

Reply:
`compressor[ZSTD] `will be `null`, so it wont reach the switch. Anyways, made the try catch to only surround `compress()`.